colossalai/fx/tracer/experimental.py

```diff
@@ -263,11 +281,42 @@ def _check_arg_name_valid(names):
         self.concrete_args = concrete_args
         self.meta_args = meta_args
 
-        with _TorchTensorOverride(self):
+        with _TorchTensorOverride(self), self.trace_activation_checkpoint(enabled=self.trace_act_ckpt):
             self.graph = super().trace(root, concrete_args=concrete_args)
         self.graph.lint()
         return self.graph
 
+
+    @contextmanager
+    def trace_activation_checkpoint(self, enabled: bool):
+        if enabled:
+            orig_ckpt_func = torch.utils.checkpoint.CheckpointFunction
+
+            class PatchedCheckpointFunction(torch.autograd.Function):
+
+                @staticmethod
+                def forward(ctx, run_function, preserve_rng_state, *args):
+                    # signal that the current tracing occurs within activaton checkpoint part
+                    self.inside_torch_checkpoint_func = True
```

Comment:
now this flag should be a fifo list.
sth like [outer_ckpt_region, inner_ckpt_region, ...]
before entering a ckpt func, push the region idx
after exiting, pop the region idx